### PR TITLE
GH-1688: Base key regression test

### PR DESCRIPTION
**Pull request by mpapierski**
Ref #1688 

This commit adds a regression test, which applied on top of `release-1.2.0` fails, but passes on `dev` as the issue is already fixed by commit aca6a6f822722cd50d877c33aeeca4fac8f02e8a

This test exercises all Stored* variant of `ExecutableDeployItem` to ensure that computed `base_key` always refers to either `Account` or a `Contract` but never a `ContractPackage`.

### CODE REVIEW

File: execution_engine_testing/tests/src/test/regression/gh_1688.rs [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        "expected {} in {:?}",
+        NEW_KEY_NAME,
+        contract.named_keys()
+    );
+    assert!(
+        !account.named_keys().contains_key(NEW_KEY_NAME),
+        "unexpected {} in {:?}",
+        NEW_KEY_NAME,
+        contract.named_keys()
+    );
```
Comment (goral09) [inline execution_engine_testing/tests/src/test/regression/gh_1688.rs:224]: These lines seem to be duplicated across all tests – is that correct? If so, can we extract it?

Comment (darthsiroftardis) [reply execution_engine_testing/tests/src/test/regression/gh_1688.rs:224]: I agree, the difference seems to be in how the deploy is created, we could extract it out and build a certain deploy based on a flag?

**Review by darthsiroftardis — APPROVED** [review]
LGTM!

File: execution_engine_testing/tests/src/test/regression/gh_1688.rs (hunk 1 above)
Comment (mpapierski) [reply execution_engine_testing/tests/src/test/regression/gh_1688.rs]: Fair point - did this in 89bf83740fc6694e1fade4ec3563fe7103a6226b

**Review by EdHastingsCasperLabs — APPROVED** [review]

**Pull request merged**